File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteDuplicates(head *ListNode) *ListNode {
    
    current := head
    var previous *ListNode
    dict := make(map[int]bool)

    for current != nil {
        if _, isExist := dict[current.Val]; isExist {
            previous.Next = current.Next
        } else {
            dict[current.Val] = true
            previous = current
        }
        current = current.Next
    }

    return head
}

File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        
        current = head
        prev = None
        saved_data = {}

        while current is not None:
            if current.val in saved_data:
                prev.next = current.next
            else:
                saved_data[current.val] = True
                prev = current
            current = current.next

        return head
        